include/mbedtls/bignum.h

```diff
@@ -747,7 +748,8 @@ int mbedtls_mpi_is_prime( const mbedtls_mpi *X,
  * mbedtls_mpi_gen_prime().
  */
 typedef enum {
-    MBEDTLS_MPI_GEN_PRIME_FLAG_DH = 0x0001,      /**< (X-1)/2 is prime too */
+    MBEDTLS_MPI_GEN_PRIME_FLAG_DH = 0x0001,     /**< (X-1)/2 is prime too */
+    MBEDTLS_MPI_GEN_PRIME_CHECK_TRNG = 0x0004   /**< check trng validity Note: 1% false positive probability */
```

Comment:
This flag has nothing to do with the platform's TRNG if any. The MPI module never interacts directly with any TRNG. This flag is about detecting the **RNG**. The documentation will also need to explain what “check validity” means.

Reply:
Same holds for the error name `MBEDTLS_ERR_MPI_TRNG_POSSIBLY_FAULTY`.

Reply:
I definitely don't agree with the current design here. A 1% probability of false positive is far too high. The probability needs to be low enough that a failure indicates that something is wrong. It doesn't make sense for the caller to retry: there's already a retry loop inside the bignum module.

Also “if RNG is valid please retry” doesn't make sense for another reason: how would the caller know whether the RNG is valid? The only information it has is that we've just reported that the RNG is not valid.

Where does this 1% number come from?

Reply:
I'm going to refer to @yanesca (please chime in) , in the original Jira ticket he wrote: 
The probability of a false positive is 1%. This is the consequence of the FIPS boundary, we can make it optional if we feel that it is too high in a general use case.

I raised concerns that this was too high, and would lead to false positives, but IIRC Janos instructed me to proceed with this initially.

Reply:
@yanesca : as we previously discussed this check is currently not optional when generating RSA keys (if calling the MPI api directly you can choose to opt in or not). how would you like to proceed with this? Would you like to add a build flag?

Reply:
I tried to find my calculations that led to the 1% number and I couldn't find them. So I redid the math and it turns out that the false positive probability is more like 0.07%.

Reply:
@NirSonnenschein I think that it would be the best to deprecate `mbedtls_rsa_gen_key()` and introduce a `mbedtls_rsa_gen_key_ext()`, that has a `flags` input parameter and enables accessing not just this, but all the other upcoming prime generation options.

Reply:
@gilles-peskine-arm : 
do you approve of @yanesca  latest suggestion:
 think that it would be the best to deprecate mbedtls_rsa_gen_key() and introduce a mbedtls_rsa_gen_key_ext(), that has a flags input parameter and enables accessing not just this, but all the other upcoming prime generation options. If so i will make this change.

I asssume in this case the behavior of the depracated mbedtls_rsa_gen_key funciton will remain unchanged and the new function will pass in the flags it is give.

Reply:
Even 0.07% would be far too high, but fortunately I think it's not that bad, see https://github.com/ARMmbed/mbedtls/pull/2036/files#r222381825